ce/ce/archivers/git.ts

```diff
@@ -129,19 +127,123 @@ export class Git {
   async reset(events: Partial<InstallEvents>, options: { commit?: string, recurse?: boolean, hard?: boolean } = {}) {
     const result = await execute(this.#toolPath, [
       '-C',
-      this.#targetFolder,
+      this.#targetFolder.fsPath,
       'reset',
       options.commit ? options.commit : '',
       options.recurse ? '--recurse-submodules' : '',
       options.hard ? '--hard' : ''
     ], {
-      env: this.#environment
+      env: this.#environment,
+      cwd: this.#targetFolder.fsPath,
+      onStdErrData: (chunk) => {
+        // generate progress events
+        const regex = /\s([0-9]*?)%/;
+        chunk.toString().split(/^/gim).map((x: string) => x.trim()).filter((each: any) => each).forEach((line: string) => {
+          const match_array = line.match(regex);
+          if (match_array !== null) {
+            events.heartbeat?.(line.trim());
+          }
+        });
+      }
     });
+    return result.code === 0 ? true : false;
+  }
+
+
+  /**
+   * Initializes a folder on disk to be a git repository
+   * @returns true if the initialization was successful, false otherwise.
+   */
+  async init() {
+    if (! await this.#targetFolder.exists()) {
+      await this.#targetFolder.createDirectory();
+    }
 
-    if (result.code) {
-      return false;
+    if (! await this.#targetFolder.isDirectory()) {
+      throw new Error(`${this.#targetFolder.fsPath} is not a directory.`);
     }
 
-    return true;
+    const result = await execute(this.#toolPath, ['init'], {
+      env: this.#environment,
+      cwd: this.#targetFolder.fsPath
+    });
+
+    return result.code === 0 ? true : false;
+  }
+
+  /**
+   * Adds a remote location to the git repo.
+   * @param name the name of the remote to add.
+   * @param location the location of the remote to add.
+   * @returns true if the addition was successful, false otherwise.
+   */
+  async addRemote(name: string, location: Uri) {
+    const result = await execute(this.#toolPath, [
+      '-C',
+      this.#targetFolder.fsPath,
+      'remote',
```

Comment:
It may be worth adding a comment here that you're adding a remote and pulling that because you want submodules to work, unlike something like

https://github.com/microsoft/vcpkg-tool/blob/35c8e299dfd7b75c4b458204edc3290cdb325e6b/src/vcpkg/vcpkgpaths.cpp#L1149

Reply:
ok

Reply:
added to where it's used (`ce/ce/installers/git.ts`)